vortex-scan/src/lib.rs

```diff
@@ -64,21 +63,39 @@ impl Scan {
     }
 
     /// Compute the result dtype of the scan given the input dtype.
-    pub fn result_dtype(&self, dtype: &DType) -> VortexResult<DType> {
-        Ok(self
-            .projection
-            .evaluate(&Canonical::empty(dtype)?.into_array())?
-            .dtype()
-            .clone())
+    pub fn result_dtype(&self) -> &DType {
+        &self.projection_dtype
     }
 
     /// Instantiate a new scan for a specific range. The range scan will share statistics with this
     /// parent scan in order to optimize future range scans.
-    pub fn range_scan(self: Arc<Self>, row_mask: RowMask) -> VortexResult<RangeScan> {
-        Ok(RangeScan::new(
+    pub fn range_scanner(self: Arc<Self>, row_mask: RowMask) -> VortexResult<RangeScanner> {
+        Ok(RangeScanner::new(
             self,
             row_mask.begin(),
             row_mask.into_filter_mask()?,
         ))
     }
 }
+
+pub struct Scan {
```

Comment:
This struct should live in vortex-file I think, as part of its API.

Reply:
Moved